core/interfaces/i_plugin.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a plugin.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IPlugin');
```

Comment:
I think we should have one more interface hierarchically that's above this that is something like `IRegistrable`. This would be a shallow interface in that it doesn't have any methods you need to implement, but it's a head nod to any object such as Field / Insertion marker / Renderer that we expect users to register. That can also be the abstract type of the object being registered in `Blockly.registry.register`